repl/src/main/scala/ammonite/repl/Storage.scala

```diff
@@ -57,6 +58,57 @@ object Storage{
       }
     }
 
+    def compileCacheSave(tag: String, classFiles: Traversable[(String, Array[Byte])], imports: Seq[ImportData]): Unit = {
+      val cacheDir = new File(dir + s"/compileCache/$tag")
+      if(!cacheDir.exists){
+        cacheDir.mkdirs()
+        val metadata = upickle.default.write(imports)
+        val metadataFw = new FileWriter(cacheDir + "/metadata.json")
+        metadataFw.write(metadata)
+        metadataFw.flush()
+        classFiles.foreach{ case (name, bytes) =>
+          val fos = new java.io.FileOutputStream(cacheDir + s"/$name.class")
+          fos.write(bytes)
+          fos.flush()
+        }
+      }
+    }
+
+    def compileCacheLoad(tag: String): Option[(Traversable[(String, Array[Byte])], Seq[ImportData])] = {
+      val cacheDir = new File(dir + s"/compileCache/$tag")
+      if(!cacheDir.exists) None
+      else for{
+        metadataJson <- try{
```

Comment:
We seem to be doing this "try, catch exception, convert to Option" thing a lot; at least three times in this code. Could we try breaking that out into a `catching[T, E](t: T): E` helper?